src/mqtt_client.rs

```diff
@@ -629,13 +629,8 @@ fn handle_listing_request(
         None
     };
 
-    if response_topic.len() > MAX_TOPIC_LENGTH {
-        return Err("Response topic too long");
-    }
-
     Ok(ListCache {
-        // `TryFrom` is misleading as it just calls `Into`
-        topic: String::from(response_topic),
```

Comment:
Didn't we just change this?

Reply:
Yes. It was actually a bug before. Now it needs to be try from

Reply:
Ah sorry when I saw this, I thought we were reverting a change we just made. I looked at the diff backwards